FMK-1004 : libhugetlbfs is failing on Fedora30 (cont)

idorax opened the pull request:
The previous fix (commit id: 8ad4422) introduced a bug that 32-bit packages are missing if we run it on 64-bit x86 system. Hence, I enhanced the Makefile to support to get the required packages according to processor type (`uname -p`) of system under test.

rasibley commented:
@cki-bot please test

idorax commented:
The lastest fix for Makefile v.s. original Makefile
```patch
$ git diff -r 2e81598 6160bc6
diff --git a/vm/hugepage/libhugetlbfs/Makefile b/vm/hugepage/libhugetlbfs/Makefile
index 22a1efe..bc101a8 100644
--- a/vm/hugepage/libhugetlbfs/Makefile
+++ b/vm/hugepage/libhugetlbfs/Makefile
@@ -107,6 +107,43 @@ run: testfullbuild
 clean:
        rm -rf *~ $(TARGET)
 
+# Get packages required to build C files according to the the processor type
+ARCH = $(shell uname -p)
+ifeq ($(ARCH), x86_64)
+       ARCH32 = i686
+       ARCH64 = x86_64
+else ifeq ($(ARCH), ppc64)
+       ARCH32 =
+       ARCH64 = ppc64
+else ifeq ($(ARCH), ppc64le)
+       ARCH32 =
+       ARCH64 = ppc64le
+else ifeq ($(ARCH), s390x)
+       ARCH32 =
+       ARCH64 = s390x
+else ifeq ($(ARCH), aarch64)
+       ARCH32 =
+       ARCH64 = aarch64
+else
+       ARCH32 = $(ARCH)
+       ARCH64 =
+endif
+ifneq ($(ARCH32), )
+       CPKGS32 = libgcc.$(ARCH32)
+       CPKGS32 += glibc.$(ARCH32)
+       CPKGS32 += glibc-headers.$(ARCH32)
+       CPKGS32 += glibc-devel.$(ARCH32)
+       CPKGS32 += glibc-static.$(ARCH32)
+endif
+ifneq ($(ARCH64), )
+       CPKGS64 = libgcc.$(ARCH64)
+       CPKGS64 += glibc.$(ARCH64)
+       CPKGS64 += glibc-headers.$(ARCH64)
+       CPKGS64 += glibc-devel.$(ARCH64)
+       CPKGS64 += glibc-static.$(ARCH64)
+endif
+CPKGS = $(CPKGS32) $(CPKGS64)
+
 include /usr/share/rhts/lib/rhts-make.include
 
 $(METADATA): Makefile
@@ -117,15 +154,10 @@ $(METADATA): Makefile
        @echo "Description:     Test libhugetlbfs with upstream testsuite" >> $(METADATA)
        @echo "TestTime:        40m" >> $(METADATA)
        @echo "RunFor:          $(PACKAGE_NAME) $(PACKAGE_NAME)-utils" >> $(METADATA)
-       @echo "Requires:        @development \
-               libgcc.i686    glibc-devel.i686    glibc-static.i686 \
-               libgcc.x86_64  glibc-devel.x86_64  glibc-static.x86_64 \
-               libgcc.ppc     glibc-devel.ppc     glibc-static.ppc \
-               libgcc.ppc64   glibc-devel.ppc64   glibc-static.ppc64 \
-               libgcc.ppc64le glibc-devel.ppc64le glibc-static.ppc64le \
-               libgcc.s390    glibc-devel.s390    glibc-static.s390 \
-               libgcc.s390x   glibc-devel.s390x   glibc-static.s390x \
-               libgcc.aarch64 glibc-devel.aarch64 glibc-static.aarch64" >> $(METADATA)
+       @echo "Requires:        gcc" >> $(METADATA)
+       @echo "Requires:        wget" >> $(METADATA)
+       @echo "Requires:        patch" >> $(METADATA)
+       @echo "Requires:        $(CPKGS)" >> $(METADATA)
        @echo "Priority:        Normal" >> $(METADATA)
        @echo "License:         GPL-v2" >> $(METADATA)
        @echo "Confidential:    no" >> $(METADATA)
```

idorax commented:
@cki-bot, the failure is not introduced by this fix.

emanchado commented:
Looks good, and it's clear that the failure is unrelated. Should I just merge?

idorax commented:
Hi @emanchado, after discuss with the test maintainer @wangli5665, we think it's simple to revert the  Makefile and just add 'wget patch gcc libgcc' to Requires part. Please help to review again, thanks!

emanchado commented:
Hi! I'm a bit confused about this PR now. You say that it was better to revert the Makefile and add `wget patch gcc` to the Requires part? But in the current diff for this PR, what I see is `@development` and combinations of `libgcc`, `glibc-devel` and `glibc-static` in Requires. Also, `wget`, `patch`, and `gcc` seem to be already there?

Is that correct? This might be easier to clarify on IRC. I'm "esteban" there if you prefer that.

idorax commented:
Hi @emanchado, before file PR#380, I filed PR#379 to fix the failure. In PR#379, 
```patch
diff --git a/vm/hugepage/libhugetlbfs/Makefile b/vm/hugepage/libhugetlbfs/Makefile
index 22a1efe..f32b20a 100644
--- a/vm/hugepage/libhugetlbfs/Makefile
+++ b/vm/hugepage/libhugetlbfs/Makefile
@@ -117,15 +117,15 @@ $(METADATA): Makefile
        @echo "Description:     Test libhugetlbfs with upstream testsuite" >> $(METADATA)
        @echo "TestTime:        40m" >> $(METADATA)
        @echo "RunFor:          $(PACKAGE_NAME) $(PACKAGE_NAME)-utils" >> $(METADATA)
-       @echo "Requires:        @development \
-               libgcc.i686    glibc-devel.i686    glibc-static.i686 \
-               libgcc.x86_64  glibc-devel.x86_64  glibc-static.x86_64 \
-               libgcc.ppc     glibc-devel.ppc     glibc-static.ppc \
-               libgcc.ppc64   glibc-devel.ppc64   glibc-static.ppc64 \
-               libgcc.ppc64le glibc-devel.ppc64le glibc-static.ppc64le \
-               libgcc.s390    glibc-devel.s390    glibc-static.s390 \
-               libgcc.s390x   glibc-devel.s390x   glibc-static.s390x \
-               libgcc.aarch64 glibc-devel.aarch64 glibc-static.aarch64" >> $(METADATA)
+       @echo "Requires:        glibc" >> $(METADATA)
+       @echo "Requires:        glibc-common" >> $(METADATA)
+       @echo "Requires:        glibc-headers" >> $(METADATA)
+       @echo "Requires:        glibc-devel" >> $(METADATA)
+       @echo "Requires:        glibc-static" >> $(METADATA)
+       @echo "Requires:        wget" >> $(METADATA)
+       @echo "Requires:        patch" >> $(METADATA)
+       @echo "Requires:        libgcc" >> $(METADATA)
+       @echo "Requires:        gcc" >> $(METADATA)
        @echo "Priority:        Normal" >> $(METADATA)
        @echo "License:         GPL-v2" >> $(METADATA)
        @echo "Confidential:    no" >> $(METADATA)

```
It introduced a bug that 32-bit packages (libgcc.i686, glibc-devel.i686, glibc-static.i686) are missing if we test the case on x86_64.

So, I filed this PR to fix the bug I introduced in PR#379. My original fix was a bit complicated because I don't think we should care about those 32-bit packages (e.g. libgcc.ppc, glibc-devel.ppc, ...) if we just test it on x86. After discussing with Li Wang, we agree that the simple/rough way works fine. Hence, I reverted Makefile to its original state before PR#379 then added `wget patch gcc` to its Requires.

idorax commented:
Without PR#379, the fix should looks like:
```patch
diff --git a/vm/hugepage/libhugetlbfs/Makefile b/vm/hugepage/libhugetlbfs/Makefile
index 22a1efe..a90e215 100644
--- a/vm/hugepage/libhugetlbfs/Makefile
+++ b/vm/hugepage/libhugetlbfs/Makefile
@@ -126,6 +126,9 @@ $(METADATA): Makefile
                libgcc.s390    glibc-devel.s390    glibc-static.s390 \
                libgcc.s390x   glibc-devel.s390x   glibc-static.s390x \
                libgcc.aarch64 glibc-devel.aarch64 glibc-static.aarch64" >> $(METADATA)
+       @echo "Requires:        wget" >> $(METADATA)
+       @echo "Requires:        patch" >> $(METADATA)
+       @echo "Requires:        gcc" >> $(METADATA)
        @echo "Priority:        Normal" >> $(METADATA)
        @echo "License:         GPL-v2" >> $(METADATA)
        @echo "Confidential:    no" >> $(METADATA)
```

emanchado commented:
Aha, I see! Sorry for the confusion. Looks good, then!

idorax commented:
Thank you, @emanchado !
